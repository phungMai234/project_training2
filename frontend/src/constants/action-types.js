const asyncActionType = (type) => ({
  PENDING: `${type}_PENDING`,
  SUCCESS: `${type}_SUCCESS`,
  ERROR: `${type}_ERROR`,
});
export const GET_TYPES = asyncActionType('GET_TYPES');
export const GET_RANKING = asyncActionType('GET_RANKING');
