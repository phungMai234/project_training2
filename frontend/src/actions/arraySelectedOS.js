import {GET_ARRAY_SELECTED_OS} from "../constants/action-types";

const getArraySelectedOSSuccess = (data) => ({
  type: GET_ARRAY_SELECTED_OS.SUCCESS,
  payload: data
})
const getArraySelectedOSError = (message) => ({
  type: GET_ARRAY_SELECTED_OS.ERROR,
  message: message
})
export const getArraySelectedOS = (arrayOS) => {
  return dispatch => {
    if (!arrayOS) {
      dispatch(getArraySelectedOSError("Select at least OS"));
    }
    dispatch(getArraySelectedOSSuccess(arrayOS))
  }
}
