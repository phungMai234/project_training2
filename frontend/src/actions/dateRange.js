import {GET_DATE_RANGE} from "../constants/action-types";

const getDateRangeSuccess = (data) => ({
  type: GET_DATE_RANGE.SUCCESS,
  payload: data
})
const getDateRangeError = (message) => ({
  type: GET_DATE_RANGE.ERROR,
  message: message
})
export const getDateRange = (dateRange) => {
  return dispatch => {
    if (!dateRange.startDate || !dateRange.endDate) {
      dispatch(getDateRangeError("Missing date range"));
    }
    dispatch(getDateRangeSuccess(dateRange))
  }
}
