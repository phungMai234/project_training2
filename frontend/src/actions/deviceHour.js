import {GET_HOUR} from "../constants/action-types";
import axios from "axios"

const getHourRequest = () => ({
  type: GET_HOUR.PENDING
})
const getHourSuccess = (data) => ({
  type: GET_HOUR.SUCCESS,
  payload: data
})
const getHourError = (message) => ({
  type: GET_HOUR.ERROR,
  message: message
})
export const getHour = (dateRange) => {
  return dispatch => {
    dispatch(getHourRequest());
    return axios.get(`http://localhost:3001/request/device_hour?startDate=${dateRange.startDate}&endDate=${dateRange.endDate}`)
      .then(res => {
        return dispatch(getHourSuccess(res.data))
      })
      .catch(err => dispatch(getHourError(err.message)));
  }
}
