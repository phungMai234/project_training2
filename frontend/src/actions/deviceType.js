import {GET_DEVICE_TYPE} from "../constants/action-types";
<<<<<<< HEAD
import {getTypes} from "../api/deviceType-api";

export const  getDeviceType = async (dispatch) => {
    const res = await getTypes();
    return dispatch({
      type: GET_DEVICE_TYPE,
      res
    })
=======

export const getDeviceType = (types) =>{
    return {
        type: GET_DEVICE_TYPE,
        types: types
    }
>>>>>>> 3192a90e76a73f76f3377cd371d1f1101668d4e5
}
