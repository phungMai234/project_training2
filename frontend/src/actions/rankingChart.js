import {GET_RANKING} from "../constants/action-types";
import axios from "axios";

const getRankingRequest = () => ({
  type: GET_RANKING.PENDING
})
const getRankingSuccess = (data) => ({
  type: GET_RANKING.SUCCESS,
  payload: data
})
const getRankingError = (message) => ({
  type: GET_RANKING.ERROR,
  message: message
})
export const getRanking = (dateRange) => {
  return dispatch => {
    dispatch(getRankingRequest());
    return axios.get(`http://localhost:3002/request/ranking?startDate=${dateRange.startDate}&endDate=${dateRange.endDate}`)
      .then(res => dispatch(getRankingSuccess(res.data)))
      .catch(err => dispatch(getRankingError(err.message)));
  }
}