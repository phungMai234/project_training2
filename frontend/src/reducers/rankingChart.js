import {GET_RANKING} from "../constants/action-types";

const initialState = {
  data: [],
  isFetching: false
}
export const rankingReducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_RANKING.SUCCESS: {
      return {...state, isFetching: false, data: action.payload}
    }
    case GET_RANKING.ERROR: {
      return {...state, isFetching: false, errorMessage: action.message}
    }
    case GET_RANKING.PENDING: {
      return {...state, isFetching: true}
    }
    default:
      return state;
  }
}