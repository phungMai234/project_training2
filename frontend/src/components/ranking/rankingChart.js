import React, {useEffect} from "react";
import "./ranking.css"
import RankingItem from "../rankingItem/rankingItem";
import {connect} from "react-redux";
import {getRanking} from "../../actions/rankingChart";
import {Spinner} from "../spinner/spinner";

function RankingChart(props) {
  let listLevel = props.rankings;
  let isFetching = props.isFetching;

  useEffect(() => {
    const fetchData = () => {
      return props.getRanking();
    }
    fetchData();

    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  return (
    <div className="ranking">
      <label className="text-secondary">Ranking</label>
      <div className="d-flex justify-content-between">
        <span>Day</span>
        <span>Number</span>
      </div>
      {isFetching ? <Spinner/> :
        listLevel.map((e, index) => {
          return <RankingItem key={index} day={index + 1} level={e}/>
        })
      }
    </div>

  )
}

const mapStateToProps = (state) => ({
  rankings: state.rankings.data,
  isFetching: state.rankings.isFetching
})

const mapDispatchToProps = dispatch => ({
  getRanking: () => dispatch(getRanking())
})

export default connect(mapStateToProps, mapDispatchToProps)(RankingChart);