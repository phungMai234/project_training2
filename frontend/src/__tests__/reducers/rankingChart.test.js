import {rankingReducer} from "../../reducers/rankingChart";
import {getRanking} from "../../actions/rankingChart";
import {GET_RANKING} from "../../constants/action-types";

const initialState = {
  data: [],
  isFetching: false
}
describe("ranking reducer", () => {
  it("should get ranking to empty list", () => {
    expect(rankingReducer(initialState, getRanking())).toMatchSnapshot();
  })
  it('should handle unknown actions', () => {
    expect(rankingReducer(undefined, {})).toEqual(initialState);
  });
  it("should handle loading time", () => {
    expect(rankingReducer(initialState, {type: GET_RANKING.PENDING})).toEqual({data: [], isFetching: true});
  })
  it('should handle error', (err = "error") => {
    expect(rankingReducer(initialState, {type: GET_RANKING.ERROR, message: err})).toEqual({
      data: [],
      isFetching: false,
      errorMessage: err
    });
  });
})