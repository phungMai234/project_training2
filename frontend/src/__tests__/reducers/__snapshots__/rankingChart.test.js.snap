// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ranking reducer should get ranking to empty list 1`] = `
Object {
  "data": Array [],
  "isFetching": false,
}
`;
